modules/imap/output_modules.php

```diff
@@ -383,7 +383,11 @@ protected function output() {
             $txt .= '<a class="delete_link hlink text-decoration-none" id="delete_message" href="#">'.$this->trans('Delete').'</a>';
             $txt .= '<div class="position-relative"><a class="hlink" id="copy_message" href="#">'.$this->trans('Copy').'</a><div class="move_to_location"></div></div>';
             $txt .= '<div class="position-relative"><a class="hlink" id="move_message" href="#">'.$this->trans('Move').'</a><div class="move_to_location"></div></div>';
-            $txt .= '<a class="archive_link hlink text-decoration-none" id="archive_message" href="#">'.$this->trans('Archive').'</a>';
+                        
+            $current_folder = end(explode("_", $this->html_safe($this->get('msg_list_path'))));
+            if (bin2hex('Archive') != $current_folder) {
```

Comment:
Archive might be the one or might not - we have a folder selection mechanism where people choose which folder to act as junk, sent, archive, etc. Furthermore, EWS tends to handle these folder comparisons a bit differently. Please move the check for archive folder in the relevant implementations (hm-mailbox bridge and then passing to hm-imap, hm-ews, etc. where they know the best way to get/compare the Archive folder).
Also, please see if the current folder is not already passed as a variable from handlers. If not, pass it. Getting it here from list_path can lead to problems in the future (e.g. changing list path) and is not exactly the things output modules should do.